### PR TITLE
[release/v2.14] Leader election config in ComponentSettings (#6641)

### PR DESCRIPTION
**Pull request by irozzo-1A**
This is an automated cherry-pick of #6641

/assign irozzo-1A

```release-note
Provide the possibility of configuring leader election parameters for user cluster components.
```

### CODE REVIEW

File: api/pkg/validation/cluster.go [hunk 1]
```diff
@@ -464,3 +464,22 @@ func ValidateUpdateWindow(updateWindow *kubermaticv1.UpdateWindow) error {
...
+		return fmt.Errorf("retry period seconds cannot be negative: %d", *l.RetryPeriodSeconds)
+	}
+	if lds, rds := l.LeaseDurationSeconds, l.RenewDeadlineSeconds; (lds == nil) != (rds == nil) {
+		return errors.New("leader election lease duration and renew deadline should be either both specified or unspecified")
+	}
+	if lds, rds := l.LeaseDurationSeconds, l.RenewDeadlineSeconds; lds != nil && *lds < *rds {
+		return errors.New("control plane leader election renew deadline cannot be smaller than lease duration")
+	}
+	return nil
+}
```
Comment (xmudrii) [inline api/pkg/validation/cluster.go]: We removed this function when we cherry-picked to v2.15. Should we remove it now as well?

Comment (irozzo-1A) [reply api/pkg/validation/cluster.go]: yes, indeed.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Comment by xrstf** [comment]
@irozzo-1A Do we need a new release for this?

**Comment by irozzo-1A** [comment]
> @irozzo-1A Do we need a new release for this?

Yes indeed. Do you take care? @xrstf

**Comment by xrstf** [comment]
yep, can do

**Pull request merged**